have_func: checking for rb_category_warn()... -------------------- no

"gcc -o conftest -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0/x86_64-linux -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0/ruby/backward -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0 -I. -I/home/ec2-user/.rbenv/versions/2.7.8/include    -g -O2 -fPIC conftest.c  -L. -L/home/ec2-user/.rbenv/versions/2.7.8/lib -Wl,-rpath,/home/ec2-user/.rbenv/versions/2.7.8/lib -L. -L/home/ec2-user/.rbenv/versions/2.7.8/lib  -fstack-protector-strong -rdynamic -Wl,-export-dynamic     -Wl,-rpath,/home/ec2-user/.rbenv/versions/2.7.8/lib -L/home/ec2-user/.rbenv/versions/2.7.8/lib -lruby  -lm   -lc"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: int main(int argc, char **argv)
4: {
5:   return !!argv[argc];
6: }
/* end */

"gcc -o conftest -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0/x86_64-linux -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0/ruby/backward -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0 -I. -I/home/ec2-user/.rbenv/versions/2.7.8/include    -g -O2 -fPIC conftest.c  -L. -L/home/ec2-user/.rbenv/versions/2.7.8/lib -Wl,-rpath,/home/ec2-user/.rbenv/versions/2.7.8/lib -L. -L/home/ec2-user/.rbenv/versions/2.7.8/lib  -fstack-protector-strong -rdynamic -Wl,-export-dynamic     -Wl,-rpath,/home/ec2-user/.rbenv/versions/2.7.8/lib -L/home/ec2-user/.rbenv/versions/2.7.8/lib -lruby  -lm   -lc"
conftest.c: 関数 ‘t’ 内:
conftest.c:14:57: エラー: ‘rb_category_warn’ undeclared (first use in this function); did you mean ‘rb_integer_pack’?
 int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_category_warn; return !p; }
                                                         ^~~~~~~~~~~~~~~~
                                                         rb_integer_pack
conftest.c:14:57: 備考: 未宣言の識別子は出現した各関数内で一回のみ報告されます
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11: 
12:   return !!argv[argc];
13: }
14: int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_category_warn; return !p; }
/* end */

"gcc -o conftest -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0/x86_64-linux -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0/ruby/backward -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0 -I. -I/home/ec2-user/.rbenv/versions/2.7.8/include    -g -O2 -fPIC conftest.c  -L. -L/home/ec2-user/.rbenv/versions/2.7.8/lib -Wl,-rpath,/home/ec2-user/.rbenv/versions/2.7.8/lib -L. -L/home/ec2-user/.rbenv/versions/2.7.8/lib  -fstack-protector-strong -rdynamic -Wl,-export-dynamic     -Wl,-rpath,/home/ec2-user/.rbenv/versions/2.7.8/lib -L/home/ec2-user/.rbenv/versions/2.7.8/lib -lruby  -lm   -lc"
/tmp/ccNrSqkc.o: 関数 `t' 内:
/var/www/rails-test-app/vender/path/ruby/2.7.0/gems/date-3.3.3/ext/date/conftest.c:15: `rb_category_warn' に対する定義されていない参照です
collect2: エラー: ld はステータス 1 で終了しました
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11: 
12:   return !!argv[argc];
13: }
14: extern void rb_category_warn();
15: int t(void) { rb_category_warn(); return 0; }
/* end */

--------------------

have_var: checking for timezone in time.h with  -Werror... -------------------- yes

"gcc -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0/x86_64-linux -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0/ruby/backward -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0 -I. -I/home/ec2-user/.rbenv/versions/2.7.8/include    -g -O2 -fPIC   -Werror -c conftest.c"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <time.h>
 4: 
 5: /*top*/
 6: extern int t(void);
 7: int main(int argc, char **argv)
 8: {
 9:   if (argc > 1000000) {
10:     int (* volatile tp)(void)=(int (*)(void))&t;
11:     printf("%d", (*tp)());
12:   }
13: 
14:   return !!argv[argc];
15: }
16: int t(void) { const volatile void *volatile p; p = &(&timezone)[0]; return !p; }
/* end */

--------------------

have_var: checking for altzone in time.h with  -Werror... -------------------- no

"gcc -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0/x86_64-linux -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0/ruby/backward -I/home/ec2-user/.rbenv/versions/2.7.8/include/ruby-2.7.0 -I. -I/home/ec2-user/.rbenv/versions/2.7.8/include    -g -O2 -fPIC   -Werror -c conftest.c"
conftest.c: 関数 ‘t’ 内:
conftest.c:16:55: エラー: ‘altzone’ undeclared (first use in this function); did you mean ‘timezone’?
 int t(void) { const volatile void *volatile p; p = &(&altzone)[0]; return !p; }
                                                       ^~~~~~~
                                                       timezone
conftest.c:16:55: 備考: 未宣言の識別子は出現した各関数内で一回のみ報告されます
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: #include <time.h>
 4: 
 5: /*top*/
 6: extern int t(void);
 7: int main(int argc, char **argv)
 8: {
 9:   if (argc > 1000000) {
10:     int (* volatile tp)(void)=(int (*)(void))&t;
11:     printf("%d", (*tp)());
12:   }
13: 
14:   return !!argv[argc];
15: }
16: int t(void) { const volatile void *volatile p; p = &(&altzone)[0]; return !p; }
/* end */

--------------------

